client/rest/src/main/java/org/elasticsearch/client/RestClient.java

```diff
@@ -357,11 +357,11 @@ public void cancelled() {
     private void setHeaders(HttpRequest httpRequest, Header[] requestHeaders) {
         Objects.requireNonNull(requestHeaders, "request headers must not be null");
         for (Header defaultHeader : defaultHeaders) {
-            httpRequest.setHeader(defaultHeader);
+            httpRequest.addHeader(defaultHeader);
         }
         for (Header requestHeader : requestHeaders) {
             Objects.requireNonNull(requestHeader, "request header must not be null");
-            httpRequest.setHeader(requestHeader);
+            httpRequest.addHeader(requestHeader);
```

Comment:
I'm undecided on whether we should override the default headers with the supplied one, or just append them.

My thought is that, if it's really a request-time decision, then wait until then to set it rather than _always_ setting it and expecting it to be overridden.